client/extensions/woocommerce/app/promotions/promotion-validations.js

```diff
@@ -0,0 +1,297 @@
+		return translate( 'End date cannot be in the past.' );
+	}
+}
+
+/**
+ * Validates a minimum amount.
+ * @param { String } fieldName The field name to be validated.
+ * @param { Object } promotion The promotion to be validated.
+ * @param { String } currency The currency to be used for validation.
+ * @param { bool } showEmpty True if empty fields should result in a validation error.
+ * @return { string } Returns a validation error, or undefined if none.
+ */
+export function validateMinimumAmount( fieldName, promotion, currency, showEmpty ) {
+	const { minimumAmount, maximumAmount } = promotion;
+
+	if ( isUndefined( minimumAmount ) ) {
+		// Field is not enabled.
+		return;
+	}
+
+	if ( showEmpty && isEmpty( minimumAmount ) ) {
+		return translate( 'Amount cannot be blank.' );
+	}
+
+	if ( null !== minimumAmount ) {
+		if ( 0 >= minimumAmount ) {
+			return translate( 'Amount must be greater than zero.' );
+		}
+
+		if ( null !== maximumAmount && Number( maximumAmount ) < Number( minimumAmount ) ) {
+			return translate( 'Amount must be less than maximum amount.' );
+		}
+	}
+}
+
+/**
+ * Validates a maximum amount.
+ * @param { String } fieldName The field name to be validated.
+ * @param { Object } promotion The promotion to be validated.
+ * @param { String } currency The currency to be used for validation.
+ * @param { bool } showEmpty True if empty fields should result in a validation error.
+ * @return { string } Returns a validation error, or undefined if none.
+ */
+export function validateMaximumAmount( fieldName, promotion, currency, showEmpty ) {
+	const { minimumAmount, maximumAmount } = promotion;
+
+	if ( isUndefined( maximumAmount ) ) {
+		// Field is not enabled.
+		return;
+	}
+
+	if ( showEmpty && isEmpty( maximumAmount ) ) {
+		return translate( 'Amount cannot be blank.' );
+	}
+
+	if ( null !== maximumAmount ) {
+		if ( 0 >= maximumAmount ) {
+			return translate( 'Amount must be greater than zero.' );
+		}
+
+		if ( null !== minimumAmount && Number( minimumAmount ) > Number( maximumAmount ) ) {
+			return translate( 'Amount must be greater than minimum amount.' );
+		}
+	}
+}
+
+/**
+ * Validates an overall usage limit.
+ * @param { String } fieldName The field name to be validated.
+ * @param { Object } promotion The promotion to be validated.
+ * @param { String } currency The currency to be used for validation.
+ * @param { bool } showEmpty True if empty fields should result in a validation error.
+ * @return { string } Returns a validation error, or undefined if none.
+ */
+export function validateUsageLimit( fieldName, promotion, currency, showEmpty ) {
+	const { usageLimit, usageLimitPerUser } = promotion;
+
+	if ( isUndefined( usageLimit ) ) {
+		// Field is not enabled.
+		return;
+	}
+
+	if ( showEmpty && isEmpty( usageLimit ) ) {
+		return translate( 'Number cannot be blank.' );
+	}
+
+	if ( null !== usageLimit ) {
+		if ( 0 >= usageLimit ) {
+			return translate( 'Number must be greater than zero.' );
```

Comment:
Hi! I've found a possible matching string that has already been translated **43** times:
`translate( '{#field} must be a number greater than {#min}' )` *ES Score: 8*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).